File: library/src/main/java/com/juanimoli/strictmodenotifier/detector/CustomSlowCallDetector.java
package com.juanimoli.strictmodenotifier.detector;

import com.juanimoli.strictmodenotifier.StrictModeLog;

public class CustomSlowCallDetector implements Detector {

    @Override
    public boolean detect(StrictModeLog log) {
        return log.message.contains("StrictMode$StrictModeCustomViolation");
    }
}

File: notifier/src/main/java/io/github/juanimoli/strictmode/notifier/detector/CustomSlowCallDetector.kt
package io.github.juanimoli.strictmode.notifier.detector

import io.github.juanimoli.strictmode.notifier.StrictModeLog

class CustomSlowCallDetector : Detector {
    override fun detect(log: StrictModeLog): Boolean {
        return log.message.contains("StrictMode\$StrictModeCustomViolation")
    }
}
